lib/collection/variable-scope.js

```diff
@@ -0,0 +1,130 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+    PropertyBase = require('./property-base').PropertyBase,
+    VariableList = require('./variable-list').VariableList,
+
+    /**
+     * Convert object to JSON using the base converter
+     * @private
+     * @param {Object} obj
+     * @returns {Object}
+     *
+     * @todo there are more than one instance of this function, maybe move to util or static to Property?
+     */
+    toJSON = function (obj) {
+        return PropertyBase.prototype.toJSON.call(obj);
+    },
+
+    VariableScope;
+
+_.inherit((
+    /**
+     * Create instances of environments and globals
+     *
+     * @constructor
+     * @extends {Property}
+     *
+     * @param {Object} definition
+     */
+    VariableScope = function PostmanVariableScope (definition) {
+        // in case the definition is an array (legacy format) or existing as list, we convert to actual format
+        if (_.isArray(definition) || VariableList.isVariableList(definition)) {
+            definition = { values: definition };
+        }
+
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        VariableScope.super_.call(this, definition);
+
+        var values;
+
+        // in case of empty definition simply create values and exit
+        if (!(definition && definition.values)) {
+            values = new VariableList(this);
+        }
+        // set the parent of the externally passed list to match the current
+        else if (VariableList.isVariableList(definition.values)) {
+            if (definition.values.__parent) { // cannot attach a list already part of another collection
+                throw new Error('collection: unable to set a variable-list that already belongs to another scope.');
```

Comment:
I thought of it, but that would result in unexpected behaviour. Imagine you add a variable list by reference and unexpectedly, it has been cloned. This is a usage level edge-case but could have impact. I am cloning it for now and adding an `@note` in docs. okay?